compute_sdk/globus_compute_sdk/sdk/web_client.py

```diff
@@ -55,45 +54,60 @@ def __init__(
         serializer: t.Optional[ComputeSerializer] = None,
     ):
         if function is not None:
-            if function_name is not None or function_code is not None:
+            if any((function_name, function_code, metadata)):
                 raise ValueError(
-                    "Cannot specify 'function_name' or 'function_code'"
-                    " if 'function' is provided."
+                    "`function` specified; cannot also specify `function_code`,"
+                    " `function_name`, or `metadata`"
                 )
+
+            serializer = serializer if serializer else ComputeSerializer()
             function_name = function.__name__
-            function_code = _get_packed_code(function, serializer=serializer)
+            function_code = serializer.pack_buffers([serializer.serialize(function)])
             if description is None:
                 description = inspect.getdoc(function)
+            metadata = FunctionRegistrationMetadata(
+                python_version="{}.{}.{}".format(*sys.version_info),
```

Comment:
We do this in a couple of places, for example 
``".".join(map(str, sys.version_info[:3])),``
in sdk.utils.__init__.py and 
 ```
           "python_v": "{}.{}.{}".format(
                sys.version_info.major, sys.version_info.minor, sys.version_info.micro
            ),
```
in interchange.py#124  and in test_globuscompute.py#11
perhaps we can centralize this in an utility method of its own, or there might be mismatches.  

However, since it's already decentralized, we prob don't need to do that in this PR.  That said, I think we should do it similar to at least one of the existing ways, instead of creating a 5th variation?

Reply:
Good point; as discussed we'll see who gets to this secondary PR.  A race is begun!  :muscle:

Reply:
PR #1824 created.  Heh.